presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1116,7 +1116,13 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
             tableStatistics = new PartitionStatistics(createEmptyStatistics(), ImmutableMap.of());
         }
 
-        metastore.createTable(session, table, principalPrivileges, Optional.of(writeInfo.getWritePath()), false, tableStatistics);
+        metastore.createTable(
+                session,
+                table,
+                principalPrivileges,
+                table.getPartitionColumns().isEmpty() ? Optional.of(writeInfo.getWritePath()) : Optional.empty(),
```

Comment:
what's in the write path if the table is partitioned?  If it's not meaningful maybe this should be fixed there.  If it is, I don't understand why we can ignore it for this particular case.

Reply:
@rschlussel : So there is write path and target path in `WriteInfo`.  Basically, write path is the path where data initially write to, and target path is the final location of the data after commit.

- In `STAGE_AND_MOVE_TO_TARGET_DIRECTORY` mode: write path is the staging path, target path is the .... target path :) 
- In `DIRECT_TO_TARGET_XXX` mode: write path and the target path are the same.

For in-txn read, we should use the write path as the location.

Reply:
I see now that TableAndMore already says that currentLocation is for an unpartitioned table only, so this commit is just enforcing that in someplace where that wasn't doing that before. 
 I'm curious what the failure was when it was provided incorrectly.  It looks to me like the location is only used in places that operate on unpartitioned tables.
Also, could you add a check to the TableAndMore constructor that the value is empty if the table is partitioned?

Reply:
> I see now that TableAndMore already says that currentLocation is for an unpartitioned table only, so this commit is just enforcing that in someplace where that wasn't doing that before.

Correct. 

> Also, could you add a check to the TableAndMore constructor that the value is empty if the table is partitioned?

Will do, thanks!